pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/ByteBufferSchema.java

```diff
@@ -68,6 +77,22 @@ public ByteBuffer decode(byte[] data) {
         }
     }
 
+    @Override
+    public ByteBuffer decode(ByteBuf byteBuf) {
+        if (null == byteBuf) {
+            return null;
+        } else {
+            int size = byteBuf.readableBytes();
```

Comment:
since we are returning a `ByteBuffer` back to the user, we shouldn't use a shared byte[]. You have to copy the bytes from ByteBuf and create ByteBuffer from the copied bytes.

Reply:
@congbobo184 I think this change still returns the underlying buffer. we need to copy the bytes from the ByteBuf.